intelmq/bots/collectors/xmpp/collector.py

```diff
@@ -21,48 +39,84 @@ def init(self):
             self.stop()
 
     def process(self):
-
         if self.xmpp is None:
-            self.xmpp = sleekxmpp.XMPPBot(self.parameters.xmpp_user + '@' + self.parameters.xmpp_server,
-                                self.parameters.xmpp_password,
-                                self.send_message,
-                                self.logger)
+            self.xmpp = XMPPClientBot(self.parameters.xmpp_user + '@'
+                                      + self.parameters.xmpp_server,
+                                      self.parameters.xmpp_password,
+                                      self.parameters.xmpp_room,
+                                      self.parameters.xmpp_room_nick,
+                                      self.parameters.xmpp_room_password,
+                                      self.logger)
             self.xmpp.connect(reattempt=True)
             self.xmpp.process()
 
-    def killbot(self):
-        self.xmpp.disconnect(wait=True)
-        self.logger.info("Disconnected")
-
+            # Add Handlers and register Plugins
+            self.xmpp.register_plugin('xep_0030')  # Service Discovery
+            self.xmpp.register_plugin('xep_0045')  # Multi-User Chat
+            self.xmpp.add_event_handler("message", self.log_message)
+            self.xmpp.add_event_handler("groupchat_message", self.log_message)
+
+    def stop(self):
+        self.xmpp.disconnect()
```

Comment:
Just noticed that `stop()` could also be called when `self.xmpp` is not created yet. E.g.:
```bash
2016-05-17 11:39:00,845 - xmpp-collector - INFO - XMPPCollectorBot initialized with id xmpp-collector and version 3.4.3 (default, Oct 14 2015, 20:28:29).
2016-05-17 11:39:00,846 - xmpp-collector - INFO - Bot is starting.
2016-05-17 11:39:00,846 - xmpp-collector - ERROR - Pipeline configuration: no key 'xmpp-collector'.
2016-05-17 11:39:00,846 - xmpp-collector - ERROR - Bot initialization failed.
Traceback (most recent call last):
  File "/home/r/dev/intelmq/intelmq/lib/bot.py", line 64, in __init__
    self.load_pipeline_configuration()
  File "/home/r/dev/intelmq/intelmq/lib/bot.py", line 363, in load_pipeline_configuration
    self.stop()
  File "/home/r/dev/intelmq/intelmq/bots/collectors/xmpp/collector.py", line 60, in stop
    self.xmpp.disconnect()
AttributeError: 'NoneType' object has no attribute 'disconnect'

```